rayon-core/src/future/mod.rs

```diff
@@ -82,6 +82,11 @@ pub fn new_rayon_future<'scope, F, S>(future: F, scope: S) -> RayonFuture<F::Ite
     }
 }
 
+/// Implements a wait method which should not cause race conditions.
```

Comment:
A couple of nits about this sentence:

- It assumes that the reader knows what "a wait method" is; I think it might be better to start off with text similar to [Future::wait][w]. 
- The problem with `Future::wait` is not race conditions but rather that invoking it can cause deadlocks.

I'm torn between something like:

A version of `Future::wait` that is safe for use in the Rayon threadpool.

And something more informative like:

Blocks the current thread until this future is resolved, returning the value. This function is a version of [`Future::wait`][w] that is safe for use from within the Rayon thread-pool. The `Future::wait()` function does not coordinate with Rayon's scheduler and so calling it may lead to panics or deadlocks.

[w]: https://docs.rs/futures/0.1.8/futures/future/trait.Future.html#method.wait